### PR TITLE
Feature/multihop oracle groups

### PR DESCRIPTION
**Pull request by ethereumdegen**

### CODE REVIEW

**Comment by ethereumdegen** [comment]
Need to add some tests that prove that the oracle route hopping works properly -- use LCFA tests

**Comment by ethereumdegen** [comment]
allow the pool deployer to set and continually update a 'min collateral per principal'  like LCFA

File: packages/contracts/contracts/LenderCommitmentForwarder/extensions/LenderCommitmentGroup/LenderCommitmentGroup_Smart.sol [hunk 1]
```diff
@@ -139,7 +139,8 @@ contract LenderCommitmentGroup_Smart is
 
     IUniswapPricingLibrary.PoolRouteConfig[]  public  poolOracleRoutes ;
 
-
+    //configured by the owner. If 0 , not used. 
+    uint256 public maxPrincipalPerCollateralAmount;
```
Comment (rbcp18) [inline packages/contracts/contracts/LenderCommitmentForwarder/extensions/LenderCommitmentGroup/LenderCommitmentGroup_Smart.sol:144]: can this be updated by the pool owner?

File: packages/contracts/contracts/LenderCommitmentForwarder/extensions/LenderCommitmentGroup/LenderCommitmentGroup_Smart.sol [hunk 2]
```diff
@@ -736,22 +747,77 @@ contract LenderCommitmentGroup_Smart is
...
+      
+       //uint256 scaledPoolOraclePrice = UniswapPricingLibrary.getUniswapPriceRatioForPoolRoutes(
+       //         poolOracleRoutes
+       //     ).percent(collateralRatio);
+ 
+
+        uint256 maxPrincipalPerCollateralAmount = maxPrincipalPerCollateralAmount == 0  
+                ? pairPriceWithTwapFromOracle   
+                : Math.min(
+                    pairPriceWithTwapFromOracle,
```
Comment (rbcp18) [inline packages/contracts/contracts/LenderCommitmentForwarder/extensions/LenderCommitmentGroup/LenderCommitmentGroup_Smart.sol:827]: Isn't this only the current price? ie it's not taking into the collateral ratio %. so if the price is used, it should be divided by the collateral ratio %

File: packages/contracts/contracts/LenderCommitmentForwarder/extensions/LenderCommitmentGroup/LenderCommitmentGroup_Smart.sol (hunk 1 above)
Comment (ethereumdegen) [reply packages/contracts/contracts/LenderCommitmentForwarder/extensions/LenderCommitmentGroup/LenderCommitmentGroup_Smart.sol:144]: done

**Comment by ethereumdegen** [comment]
ready for final review

**Review by rbcp18 — APPROVED** [review]

**Pull request merged**